applications/GeoMechanicsApplication/custom_workflows/write_output.cpp

```diff
@@ -193,4 +179,24 @@ WriteConditionsFlag GeoOutputWriter::GetWriteConditionsFlagFrom(const Parameters
     return to_write_conditions_flag.at(rGiDPostFlags["WriteConditionsFlag"].GetString());
 }
 
+GeoOutputWriter::GeoOutputWriter(                                     Parameters         Settings,
+                                 const std::string& rWorkingDirectory,
+                                 ModelPart&         rModelPart) :
+      mGidIO{rWorkingDirectory + "/" + Settings["output_processes"]["gid_output"].GetArrayItem(0)["Parameters"]["output_name"].GetString(),
+        GetGiDPostModeFrom(Settings["output_processes"]["gid_output"].GetArrayItem(0)["Parameters"]["postprocess_parameters"]["result_file_configuration"]["gidpost_flags"]),
+        GetMultiFileFlagFrom(Settings["output_processes"]["gid_output"].GetArrayItem(0)["Parameters"]["postprocess_parameters"]["result_file_configuration"]["gidpost_flags"]),
+        GetWriteDeformedMeshFlagFrom(Settings["output_processes"]["gid_output"].GetArrayItem(0)["Parameters"]["postprocess_parameters"]["result_file_configuration"]["gidpost_flags"]),
+        GetWriteConditionsFlagFrom(Settings["output_processes"]["gid_output"].GetArrayItem(0)["Parameters"]["postprocess_parameters"]["result_file_configuration"]["gidpost_flags"])}
+{
+    mGidIO.InitializeMesh(0.0);
+    mGidIO.WriteMesh(rModelPart.GetMesh());
+    mGidIO.FinalizeMesh();
+    mGidIO.InitializeResults(0, rModelPart.GetMesh());
```

Comment:
Only done once per stage

Reply:
It's OK for now, but I think we may want to revisit it, since the initialization and finalization are not symmetric.